src/components/Recorder/Recorder.tsx

```diff
@@ -68,6 +63,19 @@ export class Recorder extends Component<Props, State> {
         }
     }
 
+    startStream = () => {
+        const { videoStream, screenStream, doPlayback } = this.props;
+        if (!doPlayback) {
+            if (this.videoRef && videoStream) {
+                this.videoRef.srcObject = videoStream;
+            }
+
+            if (screenStream && this.screenRef) {
+                this.screenRef.srcObject = screenStream;
+            }
+        }
```

Comment:
something is wrong here by logic.
why do you call this function if you might not assign your refs?
you should avoid calling this function if there is no need to.

so please go to any place that calls this function, and change the login so it would be called if no need to.
i.e
```
componentDidMount() {
  const { doPlayback } = this.props;
  if(doPlayback){
    return;
  }
  this.startStream(); // a condition to call a function if needed
}
```

Reply:
it requied by TS because both ekements and states are optional

Reply:
I prefer to duplicate code from outside this function and check if we should call it.
you have access to `props` from anywhere inside your component - so you may just check if you need to call the designated `startStream` function.

as a programmer, if I call a function, then it should be able to run it without any conditions from within.
in your code, calling `startStream` doesn't always starts the stream...

take a look at the example that I've attached.
you should change your code accordingly from `componentDidMount` and call `startStream` if needed.

Reply:
I rather keep the validation in startStream. You can't expect every call to remember to do validation. It's modularity.

Reply:
that's my point. 
so let's create a validation function that you may call to check if you start the stream.

something like:
```
_shouldStartStream = () => {
  // your logic here
  if(condition...){
    return false;
  }
  return true;
}
```

and after this you may call it like:
```
componentDidMount() {
  if(!this._shouldStartStream()){
    return;
  }
  this.startStream(); // a condition to call a function if needed
}
```

Reply:
I still don't think the validation should be outside startStream.. I'll ask Nir to decide here